### DISCUSSION TITLE
Putting Chia "full_node" and "wallet" databases on an SSD or HDD?

### DISCUSSION THREAD

**Discussion by persianboy1** [Chia-Network/chia-blockchain#9337] [Support]
Hi everybody. I'm new in "chia farming". as you probably know, it's possible to change the default directory of Chia "full_node" and "wallet" databases from "C:\Users\username\.chia\mainnet" to another location such as a new "directory" or "disk". Given that the size of the databases increase rapidly, it should be done sooner or later if you run out of free space on drive C. 
my question is: Technically, is there any significant difference between "the performance and speed of farming operation" when the databases mentioned above are on an SSD, and when they are on a HDD? in the other words, does the higher speed of reading and writing the chia databases boost the farming operation, or it doesn't? thanks.

**Comment by rmica** [comment]
By my experience, better put the database on the fastest drive that you've got at your disposal. Slow drives only work for harvesters. Check for "Block validation time" warnings. They'll disappear when you switch to SSD.

**Reply by persianboy1** [reply]
thank for the response. I saw this error in the log file after you mentioned about it. but here
https://githubmemory.com/repo/Chia-Network/chia-blockchain/issues/9030
someone has said that "this is not an issue". in addition, we do know that: it's absolutely not important that the plots are on an SSD or on a slow 5400 rpm HDD. considering this, is it really effective to put "db"s on a fast drive to boost farming? I mean: does this work really increase my earning amount compared to when I put the "db"s on a slow HDD? is there any document to proove this?

**Reply by rmica** [reply]
I haven't evidence to support the claim that it's causing a loss of income, on the other hand it's a warning and not an error. But the warning disappearing gives me evidence that SSD is a full-node's comfort zone. My opinion is that reducing validation times leaves more margin to overcome load peaks and unfavorable spot conditions, that could eventually lead to a loss of reward. Ok, I've got two nodes, one on SSD and one on HDD, which is the only one writing block validation time warnings, and they both rewarded recently.

**Reply by rmica** [reply]
I've tested the node on an enterprise class virtual environment machine with high perf iSCSI SAN. Warnings were still there, even if at just 2.5 secs.

**Reply by rmica** [reply]
One last note: plots and the database have totally different I/O requirements. Harvester's highly indicized read only accesses are an adhoc alghoritm that performs very well even on slow drives. Node and sqlite have stricter requirements.

**Reply by persianboy1** [reply]
thanks again. your responses were so informative and perfect. Overall, I conclude that I would rather keep the databases on SSD. this way, I can at least reduce the chance of loosing reward, althouth its probability is low. personally, I did not see a noticeable difference between rewards amount in the two(db on SSD, or HDD) in the short term. but in the long term? who knows?
